internal/cmd/beta/network/update/update.go

```diff
@@ -151,17 +151,27 @@ func parseInput(p *print.Printer, cmd *cobra.Command, inputArgs []string) (*inpu
 
 func buildRequest(ctx context.Context, model *inputModel, apiClient *iaas.APIClient) iaas.ApiPartialUpdateNetworkRequest {
 	req := apiClient.PartialUpdateNetwork(ctx, model.ProjectId, model.NetworkId)
+	addressFamily := &iaas.UpdateNetworkAddressFamily{}
 
-	payload := iaas.PartialUpdateNetworkPayload{
-		Name: model.Name,
-		AddressFamily: &iaas.UpdateNetworkAddressFamily{
-			Ipv4: &iaas.UpdateNetworkIPv4{
-				Nameservers: model.IPv4DnsNameServers,
-			},
+	if model.IPv6DnsNameServers != nil {
+		addressFamily = &iaas.UpdateNetworkAddressFamily{
 			Ipv6: &iaas.V1UpdateNetworkIPv6{
 				Nameservers: model.IPv6DnsNameServers,
 			},
-		},
+		}
+	}
+
+	if model.IPv4DnsNameServers != nil {
+		addressFamily = &iaas.UpdateNetworkAddressFamily{
+			Ipv4: &iaas.UpdateNetworkIPv4{
+				Nameservers: model.IPv4DnsNameServers,
+			},
+		}
+	}
```

Comment:
This doesn't allow setting both, right? Or are they mutually exclusive?

Reply:
Otherwise I think we can do something like
```go
addressFamily := &iaas.UpdateNetworkAddressFamily{}

if model.IPv4DnsNameServers != nil {
		addressFamily.Ipv4 = &iaas.V1UpdateNetworkIPv4{
				Nameservers: model.IPv4DnsNameServers,
        },
}

if model.IPv6DnsNameServers != nil {
		addressFamily.Ipv6 = &iaas.V1UpdateNetworkIPv6{
				Nameservers: model.IPv6DnsNameServers,
        },
}
```

Reply:
as I understood from the API, in the future, it can be possible to have ipv4 and ipv6 at the same time in the network creation payload. But I will verify with the IaaS team. 

I think in the meantime, we can use your suggestion. That looks good